konnectivity-client/pkg/client/client.go

```diff
@@ -333,6 +341,9 @@ func (t *grpcTunnel) DialContext(requestCtx context.Context, protocol, address s
 		klog.V(5).InfoS("Context canceled waiting for DialResp", "ctxErr", requestCtx.Err(), "dialID", random)
 		go t.closeDial(random)
 		return nil, &dialFailure{"dial timeout, context", DialFailureContext}
+	case <-t.done:
```

Comment:
Do we really need 3 paths for we're shutting down?

Reply:
Unfortunately yes, I don't see a clean way to consolidate these. The timeout & context canceled cases could easily be collapsed, but then we wouldn't be able to distinguish between them in the logs. This `<-t.done` case is not strictly necessary (the timeout case would be hit eventually), but it makes for a cleaner shutdown if the underlying connection gets closed prematurely (outside of konnectivity-client).